sql/parser/builtins.go

```diff
@@ -48,9 +48,15 @@ var errRoundNumberDigits = errors.New("number of digits must be greater than 0")
 type typeList []reflect.Type
 
 type builtin struct {
-	// Set to typeList{} for nullary functions and to nil for varidic
-	// functions.
+	// Variadic functions have a tricky representation:
+	// - If all arguments must be of a specific type, `types` must be
+	// `typeList{<theType>}`, and `variadic` must be true. This will
+	// generate the correct type-checking error.
+	// - If all arguments must be of the same (variable) type, `types`
```

Comment:
Oh, that's what that comment meant. It might be worth regularizing (normalizing) the code more. For example, instead of a `variadic` field, we could make `typeList` an interface:

```
type argTypes []reflect.Type

type typeList interface {
  match(types argTypes) bool
}

func (a argTypes) match(types argTypes) bool {
  // Match `a` and `types`.
}

type anyType struct{}

func (anyType) match(types argTypes) bool {
  return true
}

type variadicType reflect.Type

func (v variadicType) match(types argTypes) bool {
  // Match `v` against `types`.
}
```